lib/dependencies/URLDependency.js

```diff
@@ -132,7 +140,108 @@ URLDependency.Template = class URLDependencyTemplate extends (
 
 		runtimeRequirements.add(RuntimeGlobals.require);
 
-		if (dep.relative) {
+		// Check if we need to add prefetch/preload runtime
+		const needsPrefetch = dep.prefetch !== undefined && dep.prefetch !== false;
+		const needsPreload = dep.preload !== undefined && dep.preload !== false;
+
+		if (needsPrefetch || needsPreload) {
+			// Get the module to determine asset type
+			const module = moduleGraph.getModule(dep);
+			let asType = "";
+
+			if (module) {
+				const request = /** @type {string} */ (
+					/** @type {{ request?: string }} */ (module).request || ""
+				);
+				// Determine the 'as' attribute based on file extension
+				// Reference: MDN rel=preload documentation (https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/rel/preload)
+				// Valid 'as' values: fetch, font, image, script, style, track
+				// Note: audio/video are in spec but not supported by major browsers as of 2025
+				if (/\.(png|jpe?g|gif|svg|webp|avif|bmp|ico|tiff?)$/i.test(request)) {
```

Comment:
Consider extracting this into a separate function.

Reply:
fixed in bce3205c79214b02a855179798bac7e5de8d393e